internal/auth/auth.go

```diff
@@ -13,29 +14,31 @@ import (
 
 //go:generate mockgen -source=auth.go -package=auth -destination=mock_auth.go
 
-type RegistryAuth interface {
-	GetKeyChainFromSecret(ctx context.Context, secretName, secretNamespace string) (authn.Keychain, error)
+type RegistryAuthGetter interface {
```

Comment:
How about `RegistryAuthenticator`?

Reply:
I have though about it, but it doesn't do any authentication, it only get the keychain from somewhere (in this case from a secret in the cluster), therefore, I think that `RegistryAuthGetter` is a better description of what the interface actually does.

Reply:
Fair enough.